### PR TITLE
[Azure] Add vnet to paraglider

### PR DESCRIPTION
**Pull request by J-467**
- `AttachResource` function to add a resource with a specified resourceID and namespace to the paraglider namespace by creating a peering to gateway vnet.
- Added unit tests for the AttachResource function
- Does not rename the vnet, but would tag the resource (to be implemented in another PR)
- Refactor some existing tests and functions to enhance reusability

#AttachResourceAzure

### CODE REVIEW

File: pkg/azure/plugin.go [hunk 1]
```diff
@@ -864,3 +813,41 @@ func Setup(port int, orchestratorServerAddr string) *azurePluginServer {
 	}()
 	return azureServer
 }
+
+// Add an existing Azure resource to a paraglider deployment
+func (s *azurePluginServer) AttachResource(ctx context.Context, attachResourceReq *paragliderpb.AttachResourceRequest) (*paragliderpb.AttachResourceResponse, error) {
```
Comment (smcclure20) [inline pkg/azure/plugin.go:818]: Nit: I think the `Setup()` function should be last in the file

File: pkg/azure/sdk_handler.go [hunk 2]
```diff
@@ -290,8 +290,8 @@ func (h *AzureSDKHandler) DeleteSecurityRule(ctx context.Context, nsgName string
 	return nil
 }
 
-// GetVnetAddressSpaces returns a map of location to address space for all virtual networks (VNets) with a given prefix.
-func (h *AzureSDKHandler) GetVNetsAddressSpaces(ctx context.Context, prefix string) (map[string][]string, error) {
+// Returns a map of location to address space for all virtual networks (VNets) with a given prefix.
+func (h *AzureSDKHandler) GetVnetsWithMatchingPrefixAddressSpaces(ctx context.Context, prefix string) (map[string][]string, error) {
```
Comment (smcclure20) [inline pkg/azure/sdk_handler.go:294]: This new name confuses me -- why the change?

Comment (smcclure20) [reply pkg/azure/sdk_handler.go:294]: Ah, I see. Maybe instead: `GetAllVnetAddressSpaces`?

File: pkg/azure/test_utils.go [hunk 3]
```diff
@@ -36,6 +36,8 @@ import (
 )
 
 const (
+	fakeNsgName                              = "test-nsg-name"
+	fakeNsgID                                = "a/b/" + fakeNsgName
```
Comment (smcclure20) [inline pkg/azure/test_utils.go:40]: Maybe make this properly formatted like `uriPrefix + "Microsoft.Network/networkSecurityGroups/" + name`? In fact, isn't this redundant with `validSecurityGroupID` below?

File: pkg/azure/test_utils.go [hunk 4]
```diff
@@ -428,13 +449,32 @@ func getFakeSubnet() *armnetwork.Subnet {
...
+				{
+					Name: to.Ptr(validSubnetName),
+					ID:   to.Ptr(validParagliderSubnetId),
+				},
+			},
+		},
+	}
+}
+
+func getFakeVirtualNetworkWithUnusedAddressSpace() *armnetwork.VirtualNetwork {
```
Comment (smcclure20) [inline pkg/azure/test_utils.go:469]: I assume this is for creating fake vnets for resources to attach? I think a better name might make that more clear. How about something like `getFakeUnattachedVirtualNetwork` (and then you could follow this naming convention for other relevant things if necessary.

**Review by smcclure20 — CHANGES_REQUESTED** [review]
Just a few minor comments! 

For the future though, maybe it would be good to separate refactoring/reorganizing from new functionality as much as possible in PRs just so they are a little easier to read.

File: pkg/azure/test_utils.go (hunk 3 above)
Comment (J-467) [reply pkg/azure/test_utils.go:40]: actually, it is

File: pkg/azure/test_utils.go (hunk 4 above)
Comment (J-467) [reply pkg/azure/test_utils.go:469]: Sounds good. Will fix this

File: pkg/azure/sdk_handler.go (hunk 2 above)
Comment (J-467) [reply pkg/azure/sdk_handler.go:294]: I need to work on choosing better variable and function names. I'll go with this and add function doc to it.

File: pkg/azure/plugin.go (hunk 1 above)
Comment (J-467) [reply pkg/azure/plugin.go:818]: Done

**Comment by J-467** [comment]
> Just a few minor comments!
> 
> For the future though, maybe it would be good to separate refactoring/reorganizing from new functionality as much as possible in PRs just so they are a little easier to read.

Well noted! Thanks Sarah

File: pkg/azure/plugin.go [hunk 5]
```diff
@@ -851,3 +830,24 @@ func (s *azurePluginServer) AttachResource(ctx context.Context, attachResourceRe
...
+	paragliderpb.RegisterCloudPluginServer(grpcServer, azureServer)
+	fmt.Println("Starting server on port: ", port)
+
+	go func() {
+		if err := grpcServer.Serve(lis); err != nil {
+			fmt.Println(err.Error())
+		}
+	}()
+	return azureServer
+}
```
Comment (smcclure20) [inline pkg/azure/plugin.go:853]: Nit: newline at EOF

**Review by smcclure20 — CHANGES_REQUESTED** [review]
Just one nit, otherwise looks great!

File: pkg/azure/plugin.go (hunk 5 above)
Comment (J-467) [reply pkg/azure/plugin.go:853]: Done

**Review by smcclure20 — APPROVED** [review]
